doc/source/train/examples/lightning/lightning_mnist_example.ipynb

```diff
@@ -690,51 +969,158 @@
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m │       test_accuracy       │    0.9740999937057495     │\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m └───────────────────────────┴───────────────────────────┘\n"
      ]
     }
    ],
    "source": [
-    "from ray.train.lightning import LightningPredictor\n",
+    "import ray\n",
+    "import pytorch_lightning as pl\n",
     "\n",
-    "predictor = LightningPredictor.from_checkpoint(\n",
-    "    checkpoint, MNISTClassifier, use_gpu=use_gpu\n",
+    "from pytorch_lightning.plugins.environments.lightning_environment import (\n",
+    "    LightningEnvironment,\n",
     ")\n",
+    "from ray.air.util.torch_dist import (\n",
+    "    TorchDistributedWorker,\n",
+    "    init_torch_dist_process_group,\n",
+    "    shutdown_torch_dist_process_group,\n",
+    ")\n",
+    "\n",
+    "\n",
+    "class RayEnvironment(LightningEnvironment):\n",
+    "    \"\"\"Setup Lightning DDP training environment for Ray cluster.\"\"\"\n",
+    "\n",
+    "    def world_size(self) -> int:\n",
+    "        return int(os.environ[\"WORLD_SIZE\"])\n",
+    "\n",
+    "    def global_rank(self) -> int:\n",
+    "        return int(os.environ[\"RANK\"])\n",
+    "\n",
+    "    def local_rank(self) -> int:\n",
+    "        return int(os.environ[\"LOCAL_RANK\"])\n",
     "\n",
+    "    def set_world_size(self, size: int) -> None:\n",
+    "        # Disable it since `world_size()` directly returns data from AIR session.\n",
+    "        pass\n",
     "\n",
-    "def accuracy(logits, labels):\n",
-    "    preds = np.argmax(logits, axis=1)\n",
-    "    correct_preds = np.sum(preds == labels)\n",
-    "    return correct_preds\n",
+    "    def set_global_rank(self, rank: int) -> None:\n",
+    "        # Disable it since `global_rank()` directly returns data from AIR session.\n",
+    "        pass\n",
     "\n",
+    "    def teardown(self):\n",
+    "        pass\n",
     "\n",
-    "corrects = total = 0\n",
-    "for batch in test_dataloader:\n",
-    "    inputs, labels = batch\n",
-    "    inputs, labels = inputs.numpy(), labels.numpy()\n",
-    "    logits = predictor.predict(inputs)[\"predictions\"]\n",
-    "    total += labels.size\n",
-    "    corrects += accuracy(logits, labels)\n",
     "\n",
-    "print(\"Accuracy: \", corrects / total)"
+    "num_workers = 4\n",
+    "\n",
+    "@ray.remote\n",
+    "class TestWorker(TorchDistributedWorker):\n",
+    "    def run(self):\n",
+    "        trainer = pl.Trainer(\n",
+    "            num_nodes=num_workers,\n",
+    "            accelerator=\"gpu\",\n",
+    "            strategy=\"ddp\",\n",
+    "            plugins=[RayEnvironment()],\n",
+    "        )\n",
+    "        return trainer.test(best_model, datamodule=datamodule)\n",
```

Comment:
For consistency with the single-node example above, should we just pass in `dataloaders` instead?

Reply:
Make sense. I'll change it